### PR TITLE
Native event handler

### PR DESCRIPTION
**Pull request by hps1978**
Added base frame files:
RSkEventEmitter: Provides functionalities required by Components/Modules for event handing using addListener() and removeListeners(), and API to sendEvent to JS side. 
NotificationCenter: Provides general purpose Event notification handling

### CODE REVIEW

File: ReactSkia/core_modules/RSkEventEmitter.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+* Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+*
+* Use of this source code is governed by a BSD-style license that can be
+* found in the LICENSE file.
+*/
+
+#include <algorithm>
+#include "RSkEventEmitter.h"
```
Comment (munezbn) [inline ReactSkia/core_modules/RSkEventEmitter.cpp]: Move this down

File: ReactSkia/sdk/NotificationCenter.h [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+// SOFTWARE.
+//
+/*
+* Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
```
Comment (munezbn) [inline ReactSkia/sdk/NotificationCenter.h]: I think we need to add this line next to first Copyright, in both file

File: ReactSkia/sdk/NotificationCenter.h [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        template <typename... Args>
+        void emit(std::string eventName, Args... args);
+
+};
+
+template <typename... Args>
+unsigned int NotificationCenter::addListener(std::string eventName, std::function<void (Args...)> cb)
+{
+    if (!cb)
+    {
```
Comment (munezbn) [inline ReactSkia/sdk/NotificationCenter.h]: I assume this file is taken from notificatio center and modified. Should we follow same coding style as RNS ? i.e 
if () {
}

File: ReactSkia/sdk/NotificationCenter.h (hunk 2 above)
Comment (hps1978) [reply ReactSkia/sdk/NotificationCenter.h]: do you mean only this line? but in that case the license description are different for each of them. That's the reason I put it here, separately

File: ReactSkia/core_modules/RSkEventEmitter.cpp (hunk 1 above)
Comment (hps1978) [reply ReactSkia/core_modules/RSkEventEmitter.cpp]: done

File: ReactSkia/sdk/NotificationCenter.h (hunk 2 above)
Comment (hps1978) [reply ReactSkia/sdk/NotificationCenter.h]: understood now. made changes as per your suggestion

File: ReactSkia/sdk/NotificationCenter.h (hunk 3 above)
Comment (hps1978) [reply ReactSkia/sdk/NotificationCenter.h]: done

**Review by munezbn — APPROVED** [review]

**Review by pprabhu-ng — APPROVED** [review]

**Pull request merged**